Doc/c-api/frame.rst

```diff
@@ -0,0 +1,69 @@
+.. highlight:: c
+
+Frame Objects
+-------------
+
+.. c:type:: PyFrameObject
+
+   The C structure of the objects used to describe frame objects.
+
+   The structure is not part of the C API.
+
+   .. versionchanged:: 3.11
+      The structure moved to the internal C API headers.
+
+Public members of the Python API can be get and set with the
+:c:func:`PyObject_GetAttrString` and :c:func:`PyObject_SetAttrString`
```

Comment:
Given that this is the C API docs, any reference to Python attributes is likely to be misread as referring to C struct fields, so it is best not to list the Python attributes here.

Could you list the C API functions, `PyFrame_GetLineNumber`, etc, first?
I'll add docs to the new accessors here, as I add them.

Then add a comment that Python-level attributes can be accessed by `PyObject_GetAttr[String]` and `PyObject_SetAttr[String]` with an example:

E.g. `PyObject_GetAttrString(frame, "f_lineno")` is equivalent to `PyFrame_GetLineNumber(frame)`